packages/coreutils/src/ratelimiter.ts

```diff
@@ -126,10 +129,10 @@ export class Throttler<T = any, U = any> extends RateLimiter<T, U> {
   /**
    * Throttles function invocations if one is currently in flight.
    */
-  async invoke(): Promise<T> {
+  async invoke(): Promise<T | undefined> {
     if (this.poll.state.phase !== 'invoked') {
       void this.poll.schedule({ interval: this.limit, phase: 'invoked' });
     }
-    return this.payload.promise;
+    return this.payload!.promise;
```

Comment:
@afshin Is it safe to assume that the payload is non-null here?

Reply:
Yes it is. Same as above.